src/main/habit-modal-dialogue/habit-modal-dialogue.component.ts

```diff
@@ -0,0 +1,211 @@
+  habitName!: string;
+  goal!: number;
+  frequency!: string;
+  goalValue: number = 1;
+  runningValue = 3;
+  cyclingValue = 5;
+  goalFrequency: string = AppConstants.Times;
+  waterFrequency: string = 'MI';
+  runningFrequency: string = 'Km';
+  frequencyPerPeriod: string = AppConstants.Per_Day;
+  timeOfDay: TimeOfDay[] = [TimeOfDay.Morning, TimeOfDay.Afternoon, TimeOfDay.Evening];
+  showCalendar: boolean = false;
+  selectedDate: Date = new Date();
+  startSelectedDate: Date = new Date();
+  selectedDates: Date[] = [];
+  days: string[] = [daysOfWeek.Sunday, daysOfWeek.Monday, daysOfWeek.Tuesday, daysOfWeek.Wednesday, daysOfWeek.Thursday, daysOfWeek.Friday, daysOfWeek.Saturday];
+  intervalPerDays: string = AppConstants.repeat;
+  months: string = AppConstants.months;
+  checkRepeat: string = AppConstants.days;
+  goalWater: number = 2000;
+  repeatValue!: string;
+  protected readonly TimeOfDay = TimeOfDay;
+  protected readonly daysOfWeek = daysOfWeek;
+
+  constructor(private ref: DynamicDialogRef, private displayService: DisplayService, private habitService: HabitService, private confirmationService: ConfirmationService) {
+  }
+
+  ngOnInit(): void {
+    const today = new Date();
+    const firstDateOfMonth = new Date(today.getFullYear(), today.getMonth(), 1);
+    this.selectedDates.push(firstDateOfMonth);
+  }
+
+  close() {
+    this.ref.close();
+  }
+
+  getIconSource(): string {
+    return iconMap[this.habitName] || 'assets/svg/mark.svg';
+  }
+
+  checkDays(day: string) {
+    return this.days.includes(day);
+  }
+
+  updateIntervalPerDays(selectedInterval: string) {
+    this.intervalPerDays = selectedInterval;
+    this.checkRepeat = AppConstants.interval;
+  }
+
+  toggleDay(day: string): void {
+    const index = this.days.indexOf(day);
+
+    if (index !== -1 && this.days.length > 1) {
+      // If the day is present and there is more than one day, remove it
+      this.days.splice(index, 1);
+    } else if (index === -1) {
+      // If the day is not present, add it
+      this.days.push(day);
+    }
+    this.checkRepeat = AppConstants.days;
+  }
+
+  updateGoalFrequency(selectedFrequency: string) {
+    this.goalFrequency = selectedFrequency;
+  }
+
+  getDayValue(): string {
+    const allDays = Object.values(daysOfWeek);
+    if (this.checkRepeat == AppConstants.days) {
+      if (this.areArraysEqual(this.days, allDays)) {
+        this.repeatValue = AppConstants.Daily;
+        return AppConstants.Daily;
+      } else {
+        this.repeatValue = this.days.join(' ');
+        const selectedDays = this.days.map(day => day.substring(0, 3)); // Abbreviate to three letters
+        return selectedDays.join(',');
+      }
+    } else if (this.checkRepeat == AppConstants.interval) {
+      this.repeatValue = this.intervalPerDays;
+      return this.intervalPerDays;
+    } else {
+      this.repeatValue = this.selectedDates.map(date => date.toLocaleDateString('en-US')).join(',');
+      return this.months;
+    }
+  }
```

Comment:
if possible can you refactor this lengthy function into multiple functions

Reply:
refactored